controllers/drpolicy_controller.go

```diff
@@ -222,3 +246,37 @@ func (r *DRPolicyReconciler) fetchClusterFSIDs(ctx context.Context, peer *multic
 
 	return nil
 }
+
+func (r *DRPolicyReconciler) deleteManifestWorks(ctx context.Context, drpolicy *ramenv1alpha1.DRPolicy) (ctrl.Result, error) {
```

Comment:
The way this is implemented, there will be 1 ManifestWork per ManagedCluster. But, there can be multiple DRPolicy pointing to same ManagedCluster. So, if you just delete the ManifestWork, it'll be deleted for all drpolicies not just the one being deleted.

We could add owner reference to ManifestWork and let garbagecollector clean it or we could check for all policies before deleting. What do you think?